server/src/main/java/org/opensearch/extensions/DiscoveryExtensionNode.java

```diff
@@ -61,6 +71,15 @@ public void writeTo(StreamOutput out) throws IOException {
     public DiscoveryExtensionNode(final StreamInput in) throws IOException {
         super(in);
         this.pluginInfo = new PluginInfo(in);
+        int size = in.readVInt();
+        dependencies = new ArrayList<>(size);
+        for (int i = 0; i < size; i++) {
+            dependencies.add(new ExtensionDependency(in));
```

Comment:
Do we plan to validate the version of the dependent extension before adding them in the list? How are we avoiding version conflicts between the extension and its dependent extensions?

Reply:
Good question to add to [SDK META #108](https://github.com/opensearch-project/opensearch-sdk-java/issues/108)

Reply:
There are a lot more questions about how to use this information, e.g., do we require extensions to be defined before they're referenced to prevent circular references?  Do we control the order of initialization to make sure dependencies are initialized first? How do we handle failures / network disconnects / node crashes/reboots/bounces?  All good questions beyond the scope of this PR which is just to actually reference the dependencies.